kratos/geometries/line_2d_2.h

```diff
@@ -951,8 +951,72 @@ class Line2D2 : public Geometry<TPointType>
         // parametric coordinate of intersection on current line
         const double numerator   = ( (first_point[0]-first_point_other[0])*(first_point_other[1] - second_point_other[1]) - (first_point[1]-first_point_other[1])*(first_point_other[0]-second_point_other[0]) );
         const double denominator = ( (first_point[0]-second_point[0])*(first_point_other[1] - second_point_other[1]) - (first_point[1]-second_point[1])*(first_point_other[0]-second_point_other[0]) );
-        if (std::abs(denominator) < tolerance) // this means parallel lines.
-            return false;
+        if (std::abs(denominator) < tolerance) {
+            // We have parallel lines. Lines can only intersect if they are co-linear. Check this now.
+            const double lhs = (second_point[1] - first_point[1]) * (first_point_other[0] - second_point[0]);
+            const double rhs = (first_point_other[1] - second_point[1]) * (second_point[0] - first_point[0]);
+            if (std::abs(lhs - rhs) < tolerance) {
+                // Lines are parallel and co-linear, check that at least one point of the other line is within the local line
+                const Vector AB = second_point - first_point;
```

Comment:
is this a vector or an array? (array is faster)

Reply:
Swapped to array, thanks for the tip